### PR TITLE
CFIN-291 Add automatic run of tests on pull request and before deploying

### PR DESCRIPTION
**Pull request by ncn1**
* checks.yml now not running on push to dev and main
* deploy.yml runs formatting and linting checks, and tests, before deploying

### CODE REVIEW

**Review by ChristianBeddows — APPROVED** [review]
Looks good :)

**Review by mrgibbonsuoy — APPROVED** [review]

**Review by FergusMcGlynn — APPROVED** [review]

**Review by RichardAllum — APPROVED** [review]

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -1,9 +1,5 @@
-name: Run XO and Prettier
+name: Run Prettier and XO checks, and tests
 on:
-  push:
-    branches:
-      - dev
-      - main
```
Comment (ce666-york) [inline .github/workflows/checks.yml:6]: We had a discussion about this on a previous PR and said there's no harm in doing this, and it provides further reassurance on branches so we might as well keep it in (also gives us a target for github badges in future)

Comment (ncn1) [reply .github/workflows/checks.yml:6]: But it will then run the formatting, linting and tests twice, because it's also running them in deploy.yml

Comment (ce666-york) [reply .github/workflows/checks.yml:6]: That's true, but I think duplication is inevitable with unlinked workflows. The main purpose is thinking ahead for badges so we don't need it right now, I might try and pick it up in some downtime (if someone else doesn't get there first).

**Pull request merged**